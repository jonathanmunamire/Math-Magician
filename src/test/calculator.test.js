import React from "react";
import Calculator from '../components/calculator';
import renderer from "react-test-renderer";

describe("test if the page renders components", ()=>{
    test("test if calculator is rendered", ()=>{
        const calcul = renderer
        .create(<Calculator />)
        .toJSON();

        expect(calcul).toMatchSnapshot();
    })
})