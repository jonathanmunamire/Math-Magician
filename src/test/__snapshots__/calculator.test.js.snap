// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test if the page renders components test if calculator is rendered 1`] = `
<div
  className="calculator_page"
>
  <h2>
    Let's Do Some Math
  </h2>
  <div
    className="calculator"
  >
    <div
      className="parent"
    >
      <div
        className="child output"
      >
        0
      </div>
      <button
        className="child ac"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="child"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="child"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="child"
        onClick={[Function]}
        type="button"
      >
        /
      </button>
      <button
        className="child"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="child"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="child"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="child"
        onClick={[Function]}
        type="button"
      >
        *
      </button>
      <button
        className="child"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="child"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="child"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="child"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="child"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="child"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="child"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="child plus"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="child"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="child"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="child equal"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
